### PR TITLE
Refactor training scripts, add seed and scheduler

### PR DESCRIPTION
**Pull request by daitran-moreh**
## Background

- https://moreh.atlassian.net/browse/MV-551

### PR Content

- Add seed to training scripts and remove redundant seed settings
- Refactor code
- Add scheduler in `text_to_image`'s `log_validation` function

### CODE REVIEW

File: examples/text_to_image/train_text_to_image_mlflow.py [hunk 1]
```diff
@@ -61,41 +63,39 @@
...
     logger.info("Running validation... ")
-
     pipeline = StableDiffusionPipeline.from_pretrained(
         args.pretrained_model_name_or_path,
-        vae=accelerator.unwrap_model(vae),
-        text_encoder=accelerator.unwrap_model(text_encoder),
+        vae=vae,
+        text_encoder=text_encoder,
         tokenizer=tokenizer,
         unet=accelerator.unwrap_model(unet),
```
Comment (loctxmoreh) [inline examples/text_to_image/train_text_to_image_mlflow.py:86]: Why `unet` is unwrapped but `vae` and `text_encoder` are not? 
Imo, on MAF, whether a module is wrapped or unwrapped does not matter, so why did you make this change?

File: examples/text_to_image/train_text_to_image_mlflow.py [hunk 2]
```diff
@@ -917,7 +905,7 @@ def collate_fn(examples):
                     noisy_latents = noise_scheduler.add_noise(latents, noise, timesteps)
 
                 # Get the text embedding for conditioning
-                encoder_hidden_states = text_encoder(batch["input_ids"])[0]
+                encoder_hidden_states = text_encoder(batch["input_ids"])[0].to(dtype=weight_dtype)
```
Comment (loctxmoreh) [inline examples/text_to_image/train_text_to_image_mlflow.py]: `text_encoder` is already casted to `weight_dtype`, why do you need to cast one more time here?

File: examples/textual_inversion/textual_inversion_mlflow.py [hunk 3]
```diff
@@ -143,18 +145,19 @@ def log_validation(text_encoder, tokenizer, unet, vae, args, accelerator, weight
...
-    generator = None if args.seed is None else torch.Generator(device=accelerator.device).manual_seed(args.seed)
+    generator = None if args.seed is None else torch.Generator(device=accelerator.device)
     images = []
     for _ in range(args.num_validation_images):
         with torch.autocast("cuda"):
             image = pipeline(args.validation_prompt, num_inference_steps=25, generator=generator).images[0]
+
         images.append(image)
 
     for tracker in accelerator.trackers:
```
Comment (loctxmoreh) [inline examples/textual_inversion/textual_inversion_mlflow.py:156]: [Optional] Can we add `mlflow` as a tracker?

File: examples/textual_inversion/textual_inversion_mlflow.py [hunk 4]
```diff
@@ -619,12 +608,12 @@ def main():
         diffusers.utils.logging.set_verbosity_error()
 
     # If passed along, set the training seed now.
-    if args.seed is not None:
+    if args.seed:
```
Comment (loctxmoreh) [inline examples/textual_inversion/textual_inversion_mlflow.py]: You shouldn't do this. If `args.seed == 0` then it won't set seed, but `0` is still a valid seed

File: examples/unconditional_image_generation/train_unconditional_mlflow.py [hunk 5]
```diff
@@ -677,89 +668,43 @@ def transform_images(examples):
...
-                # mlflow.log_metric('one_epoch_time', one_epoch_time)
                 break
 
         progress_bar.close()
-        
         accelerator.wait_for_everyone()
 
         # Generate sample images for visual inspection
         if accelerator.is_main_process:
-            # if epoch % args.save_images_epochs == 0 or epoch == args.num_epochs - 1:
```
Comment (loctxmoreh) [inline examples/unconditional_image_generation/train_unconditional_mlflow.py]: if `save_images_epochs` is still exposed as a cli arg then you should keep this code blocks. Otherwise it is ok to remove

**Review by loctxmoreh — APPROVED** [review]
otherwise, LGTM

File: examples/text_to_image/train_text_to_image_mlflow.py (hunk 1 above)
Comment (daitran-moreh) [reply examples/text_to_image/train_text_to_image_mlflow.py:86]: Since in this script, `vae` and `text_encoder` are kept unchanged (`.requires_grad_(False)`) while `unet` is trained. I just want to ensure the desired behavior cuda gpu

File: examples/text_to_image/train_text_to_image_mlflow.py (hunk 2 above)
Comment (daitran-moreh) [reply examples/text_to_image/train_text_to_image_mlflow.py]: You're right. It's redundant

File: examples/textual_inversion/textual_inversion_mlflow.py (hunk 3 above)
Comment (daitran-moreh) [reply examples/textual_inversion/textual_inversion_mlflow.py:156]: I think we can. I'll try it
